### ISSUE TITLE
Does Libtorch c++ convolution operation use optimization?

### ISSUE THREAD

**Issue by peterkim333** [pytorch/pytorch#117963]
### 🐛 Describe the bug

I have experience developing cuDNN and CUDA, and conducting research and development using the C#-based CNTK toolkit, Libtorch, and python.
We implement the same operations (layers) in various frameworks and also provide the same inference software.
The problem is that the C++ code constructed with cuDNN and the libtorch C++ code show different results.
By tracking all layers in feature map units, we found that the results vary from one convolution layer.
Of course, we also checked all initialization data values including gamma, beta, bias, weight and bias of batch normalization, and running mean variance.
Strangely, several layers, including the first few convolution layers, showed the same results within the two frameworks, but the convolution results suddenly changed at certain intervals.
I know that the results do not vary this much when using a deterministic algorithm.
Does libtorch c++ sometimes use other optimization techniques? Or is there a related option that I can turn on and off?
Since no one mixes multiple deep learning frameworks and languages at the same time, I can't find related questions and answers.

thank you.

### Versions

CUDA 11.8 
Libtorch 1.13

**Comment by zou3519** [comment]
Can you reproduce this using PyTorch's python API?
